# This was added to the safebrowsing prefs in
# https://chromium-review.googlesource.com/c/chromium/src/+/6898756
# Move it to content_settings since we remove safebrowsing prefs
--- a/chrome/browser/content_settings/generated_javascript_optimizer_pref.cc
+++ b/chrome/browser/content_settings/generated_javascript_optimizer_pref.cc
@@ -13,6 +13,7 @@
 #include "chrome/common/extensions/api/settings_private.h"
 #include "components/content_settings/core/common/features.h"
 #include "components/prefs/pref_service.h"
+#include "components/content_settings/core/common/pref_names.h"
 
 using extensions::api::settings_private::Enforcement;
 using extensions::api::settings_private::PrefObject;
--- a/chrome/browser/site_protection/site_familiarity_utils.cc
+++ b/chrome/browser/site_protection/site_familiarity_utils.cc
@@ -9,6 +9,7 @@
 #include "chrome/browser/profiles/profile.h"
 #include "components/content_settings/core/browser/host_content_settings_map.h"
 #include "components/content_settings/core/common/features.h"
+#include "components/content_settings/core/common/pref_names.h"
 #include "components/prefs/pref_service.h"
 #include "content/public/browser/render_frame_host.h"
 #include "content/public/browser/render_process_host.h"
--- a/components/content_settings/core/browser/content_settings_pref_provider.cc
+++ b/components/content_settings/core/browser/content_settings_pref_provider.cc
@@ -93,6 +93,7 @@ void PrefProvider::RegisterProfilePrefs(
       ContentSettingsPattern::kContentSettingsPatternVersion);
   registry->RegisterBooleanPref(prefs::kInContextCookieControlsOpened, false);
   registry->RegisterBooleanPref(kGeolocationMigrateExceptionsPref, false);
+  registry->RegisterBooleanPref(prefs::kJavascriptOptimizerBlockedForUnfamiliarSites, false);
 
   WebsiteSettingsRegistry* website_settings =
       WebsiteSettingsRegistry::GetInstance();
--- a/components/content_settings/core/common/pref_names.h
+++ b/components/content_settings/core/common/pref_names.h
@@ -277,6 +277,8 @@ inline constexpr char kDesktopSiteWindow
     "desktop_site.window_setting";
 #endif
 
+inline constexpr char kJavascriptOptimizerBlockedForUnfamiliarSites[] =
+    "safebrowsing.javascript_optimizer_blocked_for_unfamiliar_sites";
 }  // namespace prefs
 
 #endif  // COMPONENTS_CONTENT_SETTINGS_CORE_COMMON_PREF_NAMES_H_
